api/src/routes/ranks.ts

```diff
@@ -0,0 +1,287 @@
+    logger.error(error);
+    res.status(500).send("Internal Server Error");
+  }
+});
+
+/**
+ * @swagger
+ * /ranks/average:
+ *   get:
+ *     summary: Get an average of the ranks of a user on a game
+ *     tags: [Ranks]
+ *     description: Retrieve the average rank of a user on a specific game from the database.
+ *     parameters:
+ *       - in: query
+ *         name: userId
+ *         required: true
+ *         description: ID of the user to retrieve ranks for
+ *         schema:
+ *           type: integer
+ *       - in: query
+ *         name: gameId
+ *         required: true
+ *         description: ID of the game to retrieve ranks for
+ *         schema:
+ *           type: integer
+ *     responses:
+ *       200:
+ *         description: Successful response
+ *       500:
+ *         description: Internal Server Error
+ */
+
+// TODO: Add a since/until parameter.
+router.get("/average", async (req: Request, res: Response) => {
+  try {
+    const { userId, gameId } = req.query;
+
+    const result = await pool.query(
+      "SELECT * FROM user_ranks WHERE user_id = $1 AND game_id = $2",
+      [userId, gameId]
+    );
+
+    const ranks = result.rows;
+    res.json(await getAverageRank(ranks));
+  } catch (error) {
+    logger.error(error);
+    res.status(500).send("Internal Server Error");
+  }
+});
+
+/**
+ * @swagger
+ * /ranks/connections:
+ *   get:
+ *     summary: Get all ranks of a user and their connections on a game
+ *     tags: [Ranks]
+ *     description: Retrieve all ranks of a user and their connections on a specific game from the database.
+ *     parameters:
+ *       - in: query
+ *         name: userId
+ *         required: true
+ *         description: ID of the user to retrieve ranks for
+ *         schema:
+ *           type: integer
+ *       - in: query
+ *         name: gameId
+ *         required: true
+ *         description: ID of the game to retrieve ranks for
+ *         schema:
+ *           type: integer
+ *       - in: query
+ *         name: page
+ *         description: Page number for pagination
+ *         schema:
+ *           type: integer
+ *         default: 1
+ *       - in: query
+ *         name: pageSize
+ *         description: Number of logs per page
+ *         schema:
+ *           type: integer
+ *         default: 10
+ *     responses:
+ *       200:
+ *         description: Successful response
+ *       500:
+ *         description: Internal Server Error
+ */
+
+// TODO: Add a since/until parameter.
+router.get("/connections", async (req: Request, res: Response) => {
+  try {
+    const { userId, gameId, page = 1, pageSize = 10 } = req.query;
+
+    if (!paginationIsValid(page as number, pageSize as number)) {
+      return res
+        .status(400)
+        .json({ error: "Invalid page or pageSize parameters" });
+    }
+
+    // Calculate the offset for pagination
+    const offset = (Number(page) - 1) * Number(pageSize);
+
+    const userConnections = await pool.query(
```

Comment:
COuld this not be seperated out into a function that can also be used in the `/connections` endpoint
Also I believe this again lies in the `/users` endpoint - like `GET /users/{user_id}/connections/ranks`